### ISSUE TITLE
False positive assigning-non-slot (E0237) in a class with __slots__ + __setattr__

### ISSUE THREAD

**Issue by sbacchio** [PyCQA/pylint#3793]
### Steps to reproduce
```py
"Demonstrate __slots__ + __setattr__ false positive"


class Foo:
    "Simple class with slots and setattr"

    __slots__ = [
        "foo",
    ]

    def __init__(self, foo, bar):
        self.foo = foo
        self.bar = bar

    def __setattr__(self, key, value):
        if key in self.__slots__:
            getattr(type(self), key).__set__(self, value)
        else:
            print(key, value)
```

### Current behavior
```
E0237: Assigning to attribute 'bar' not defined in class slots (assigning-non-slot)
```

### Expected behavior
No error.

### pylint --version output
pylint 2.6.0
astroid 2.4.2
Python 3.7.6 (default, Jan  8 2020, 19:59:22) 
[GCC 7.3.0]

**Comment by surkova** [comment]
This looks a lot like what's happening in Flask: https://github.com/pallets/flask/issues/4020#issuecomment-846888874
